tool_quality_checks/correct_import_file.py

```diff
@@ -0,0 +1,244 @@
+    mem_datasource = DRIVER_OGR_MEM.CreateDataSource("mem{}".format(MEM_NUM))
+    MEM_NUM = MEM_NUM + 1
+    return mem_datasource
+
+
+def create_geom_transform(in_spatial_ref, out_epsg):
+    """ Creating a coordinate transformation based on two reference systems """
+    out_spatial_ref = osr.SpatialReference()
+    out_spatial_ref.ImportFromEPSG(out_epsg)
+    coordTrans = osr.CoordinateTransformation(in_spatial_ref, out_spatial_ref)
+    return coordTrans
+
+
+def multipoly2poly(in_layer, out_layer):
+
+    lost_features = []
+    layer_defn = in_layer.GetLayerDefn()
+    field_names = []
+    for n in range(layer_defn.GetFieldCount()):
+        field_names.append(layer_defn.GetFieldDefn(n).name)
+
+    for count, in_feat in enumerate(in_layer):
+        content = {}
+        for field_name in field_names:
+            content[field_name] = in_feat.GetField(field_name)
+
+        geom = in_feat.GetGeometryRef()
+        if geom == None:
+            logger.warning("FID {} has no geometry.".format(count))
+            lost_features.append(in_feat.GetFID())
+            continue
+
+        if (
+            geom.GetGeometryName() == "MULTIPOLYGON"
+            or geom.GetGeometryName() == "MULTILINESTRING"
+        ):
+            for geom_part in geom:
+                addPolygon(geom_part.ExportToWkb(), content, out_layer)
+        else:
+            addPolygon(geom.ExportToWkb(), content, out_layer)
+
+    return lost_features
+
+
+def addPolygon(simple_polygon, content, out_lyr):
+    featureDefn = out_lyr.GetLayerDefn()
+
+    polygon = ogr.CreateGeometryFromWkb(simple_polygon)
+    out_feat = ogr.Feature(featureDefn)
+    out_feat.SetGeometry(polygon)
+
+    for key, value in content.items():
+        out_feat.SetField(key, value)
+
+    out_lyr.CreateFeature(out_feat)
+
+
+def fix_geometry(geometry):
+    geom_name = geometry.GetGeometryName()
+    # check pointcount if linestring
+    if "LINESTRING" in geom_name:
+        if geometry.GetPointCount() == 1:
+            logger.warning("Geometry point count of linestring = 1")
+            return geometry, False
+        else:
+            pass
+
+    # solve self intersections
+    if "POLYGON" in geom_name:
+        try:
+            geometry = geometry.Buffer(0)
+        except Exception as e:
+            # RuntimeError: IllegalArgumentException: Points of LinearRing do not form a closed linestring
+            logger.warning(e)
+            return geometry, False
+
+    # We think this part is not used or creates an error. So just leave it out for a while
+    # # check slivers
+    # if not geometry.IsValid():
+    #     perimeter = geometry.Boundary().Length()
+    #     area = geometry.GetArea()
+
+    #     sliver = float(perimeter / area)
+
+    #     if sliver < 1:
+    #         wkt = geometry.ExportToWkt()
+    #         geometry = ogr.CreateGeometryFromWkt(wkt)
+
+    return geometry, geometry.IsValid()
+
+
+def correct(in_layer, layer_name="", epsg=28992):
```

Comment:
Van 28992 een `RD_NEW` constante maken? Dan is het even wat explicieter.

Is dit trouwens een handige? Kan het makkelijk ook Vietnam enzo aan? Als je er 42nogwattus van maakt als default weet je tenminste zeker dat je RD (en daardoor de rest) ook aankan (omdat je dat de hele tijd test).